.github/workflows/reusable-build-system-test.yml

```diff
@@ -0,0 +1,133 @@
+# Description: this is a reusable workflow that runs build on mega-apps that have all the connected components for React Native.
+#
+# See build-system-test.yml for usage.
+on:
+  workflow_call:
+    inputs:
+      dist-tag:
+        required: true
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: ci
+    strategy:
+      fail-fast: false
+      matrix:
+        framework: [react]
+        framework-version: [latest]
+        build-tool: [cra, next, vite]
+        build-tool-version: [latest]
+        pkg-manager: [npm]
+        language: [ts]
+        node-version: [18]
+        pkg-manager-version: [latest]
+        include:
```

Comment:
Doesn't need to happen in this PR, but can we replace the values nested below with a matrix? Thinking we could have a single structure that should simplify updating versions of the support targets

Reply:
Yeah mstm